### PR TITLE
DATAES-596 Use RestHighLevelClient's methods supported both in Elasti…

### PR DESCRIPTION
**Pull request by rpuch**
Switch to methods taking RequestOptions as they are more universal as per https://jira.spring.io/browse/DATAES-596

I did not add myself to the least of authors as the changes are trivial.

And I'm afraid I've forgot a dash between ticket ID and summary in the commit message. Should I reword the commit or redo it completely?

- [ v] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [ v] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [ v] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@rpuch Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=287)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=287) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@rpuch Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=287)!

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java [hunk 1]
```diff
@@ -241,7 +241,8 @@ public boolean removeAlias(AliasQuery query) {
 	public boolean createIndex(String indexName) {
 		Assert.notNull(indexName, "No index defined for Query");
 		try {
-			return client.indices().create(Requests.createIndexRequest(indexName)).isAcknowledged();
+			CreateIndexRequest indexRequest = createIndexRequest(indexName);
+			return client.indices().create(indexRequest, RequestOptions.DEFAULT).isAcknowledged();
```
[the remark this answers is not in the document]
Comment (rpuch) [reply src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java]: > Thank you for the contribution.
> 
> * I have 1 comment in the code
> * Please add yourself to the authors list
> * I changed the title of DATAES-596 to _Fix usage of deprecated Rest(HighLevel)Client API_ (I hope you don't mind), so squash the commit to the "DATAES-596 - Fix usage of deprecated Rest(HighLevel)Client API." message

@sothawo I rewrote my commit applying the requested changes and changing the commit message, but github does not allow me to push the branch. Probably I did not understand you correctly when you said that I should 'squash the commit to the' new message. Now I'm a bit stuck. Squashing means rewriting history, and github does not allow it :) How do we proceed?

[the remark this answers is not in the document]
Comment (sothawo) [reply src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java]: you squash the commits into one (search for git rebase -i) locally and the force-push your local branch to the remote origin

[the remark this answers is not in the document]
Comment (rpuch) [reply src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java]: Done

**Comment by sothawo** [comment]
Your change has been merged; thanks for your contribution!

**Pull request merged**